src/nvim/ops.c

```diff
@@ -2538,18 +2538,18 @@ static void yank_copy_line(yankreg_T *reg, struct block_def *bd, size_t y_idx)
 ///
 /// @param oap Operator arguments.
 /// @param reg The yank register used.
-static void yank_do_autocmd(oparg_T *oap, yankreg_T *reg)
+static void do_autocmd_yank(oparg_T *oap, yankreg_T *reg)
   FUNC_ATTR_NONNULL_ALL
 {
-  static bool recursive = false;
+  static bool recursive_yank = false;
```

Comment:
A static variable inside a function means that its scope is the function, but its lifetime is that of the entire program. Thus it shouldn't have such a general name to avoid name clashes in the future.

So, `do_autocmd_textyankpost()` will use `recursive_textyankpost` and `do_autocmd_dirchanged` will use `recursive_dirchanged`.